homework/Truong_Duong/memorize.js

```diff
@@ -0,0 +1,32 @@
+// Modify the below function
+function memoize(func) {
+    const cache = {};
+
+    const memorizeFn = function (a, b) {
```

Comment:
it works for this test case, but i think you should cover more case
```js
function memoize(func) {
    const cache = {};
    return function (...agrs) {
      if(cache[agrs.join('')]) return cache[agrs.join('')]
      return cache[agrs.join('')] = func.apply(this, [...agrs])
    };
}
```

Reply:
```js
function memoize(func) {
    const cache = {};
    return function(...args) {
        if (cache[args]) {
            return cache[args];
        }
        const result = func(...args);
        cache[args] = result;
        return result;
    }
}
```

This is answer from author (I am not so sure about it)